CRM/Contribute/BAO/Query.php

```diff
@@ -568,6 +568,9 @@ public static function whereClauseSingle(&$values, &$query) {
 
       case 'contribution_batch_id':
         $batches = CRM_Contribute_PseudoConstant::batch();
+        // The key 'NULL' indicates that the user is looking for contributions
+        // that are not contained in a batch, see CRM-19325.
+        $batches['NULL'] = ts('(none)');
```

Comment:
Thank you for pointing this out. I will look at this, but probably not this week.